### PR TITLE
Response

### PR DESCRIPTION
**Pull request by jackcarlisle**
# ready for review

- [x] adds response action to the offer controller

#36

### CODE REVIEW

File: lib/ev2/production/contract_type/contract_type_helpers.ex [hunk 1]
```diff
@@ -247,4 +247,58 @@ defmodule Ev2.Production.ContractType.Helpers do
     if daily, do: "DAILY " <> current, else: current
   end
 
+  def check_loan_out(project, contract, contractor, offer) do
```
Comment (JMurphyWeb) [inline lib/ev2/production/contract_type/contract_type_helpers.ex]: Don't we already have this function? I think it should stay in here and be imported into the relevant view

File: lib/ev2/production/document/document_api.ex [hunk 2]
```diff
@@ -110,6 +110,66 @@ defmodule Ev2.Production.DocumentAPI do
...
+          project
+          |> project_documents()
+          |> Repo.all()
+          |> Enum.filter(fn document -> document.category == "Form" end)
+        documents =
+          contract_documents ++ form_documents
+          |> sort_documents()
+      end
+
+      defp sort_documents(documents) do
```
Comment (JMurphyWeb) [inline lib/ev2/production/document/document_api.ex]: Why do we do this sorting? And what does it sort it into?

File: lib/ev2/production/offer/offer_api.ex [hunk 3]
```diff
@@ -95,12 +123,31 @@ defmodule Ev2.Production.OfferAPI do
...
 
+      @doc """
+      Updates an offer's contract type.
+
+      ## Examples
+
+          iex> update_offer(offer, %{field: new_value})
+          {:ok, %Offer{}}
+
+          iex> update_offer(offer, %{field: bad_value})
```
Comment (JMurphyWeb) [inline lib/ev2/production/offer/offer_api.ex]: remove this documentation and the {:ok, Offer{}} above, and update to include the !

File: lib/ev2/production/offer/offer_api.ex [hunk 4]
```diff
@@ -119,6 +166,24 @@ defmodule Ev2.Production.OfferAPI do
...
+      @doc """
+      Updates an offer's response.
+
+      ## Examples
+
+          iex> update_offer_response!(offer, %{field: new_value})
+          {:ok, %Offer{}}
+
+          iex> update_offer_response!(offer, %{field: bad_value})
+          {:error, %Ecto.Changeset{}}
```
Comment (JMurphyWeb) [inline lib/ev2/production/offer/offer_api.ex]: Same here

File: lib/ev2/production/document/document_api.ex (hunk 2 above)
Comment (jackcarlisle) [reply lib/ev2/production/document/document_api.ex]: So i think this was to get all of the initial contracts up at the top. Think this relates to what Becca was talking about. Mustn't be 100% working then, I'll take a look

Comment (jackcarlisle) [reply lib/ev2/production/document/document_api.ex]: Should sort them in reverse order with START FORM first

File: lib/ev2_web/controllers/offer_controller.ex [hunk 5]
```diff
@@ -199,6 +202,119 @@ defmodule Ev2Web.OfferController do
...
+            # )
+            conn
+            |> put_flash(:info, "Congratulations, you have accepted this offer")
+            |> redirect(to: project_offer_path(conn, :show, project, offer))
+        end
+    end
+  # end
+  end
+
+  defp get_contractor(email) do
```
[the remark this answers is not in the document]
Comment (JMurphyWeb) [reply lib/ev2_web/controllers/offer_controller.ex]: Just checking this has been deleted if its not used anymore.

**Pull request merged**